src/main/java/br/com/caelum/comidinhas/handler/validation/ValidationErrorHandler.java

```diff
@@ -0,0 +1,49 @@
+package br.com.caelum.comidinhas.handler.validation;
+
+import br.com.caelum.comidinhas.exception.RelationshipNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.*;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestControllerAdvice
+public class ValidationErrorHandler {
+    @Autowired
+    private MessageSource messageSource;
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ValidationErrorsOutputDto handleValidationErrorArgumentNotValid(MethodArgumentNotValidException exception) {
+
+        List<ObjectError> globalErrors = exception.getBindingResult().getGlobalErrors();
+        List<FieldError> fieldErrors = exception.getBindingResult().getFieldErrors();
+
+        ValidationErrorsOutputDto validationErrors = new ValidationErrorsOutputDto();
+
+        globalErrors.forEach(error -> validationErrors.addError(getErrorMessage(error)));
+
+        fieldErrors.forEach(error -> {
+            String errorMessage = getErrorMessage(error);
+            validationErrors.addFieldError(error.getField(), errorMessage);
+        });
+
+        return validationErrors;
+    }
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(RelationshipNotFoundException.class)
+    public ValidationErrorsOutputDto handleValidationnErrorIllegalArgument(RelationshipNotFoundException exception) {
+        ValidationErrorsOutputDto validationErrors = new ValidationErrorsOutputDto();
+        validationErrors.addError(exception.getMessage());
+        return validationErrors;
+    }
```

Comment:
Corrigindo typo no nome do método.

Método não precisa ser público.

Favorecemos o uso do modificador de acesso mais restrito.



```suggestion
    ValidationErrorsOutputDto handleValidationErrorIllegalArgument(RelationshipNotFoundException exception) {
        ValidationErrorsOutputDto validationErrors = new ValidationErrorsOutputDto();
        validationErrors.addError(exception.getMessage());
        return validationErrors;
    }
```